routes/survey.ts

```diff
@@ -0,0 +1,122 @@
+import {Db_Wrapper, env_config} from "../config"
+const express = require('express');
+import { parseCSV, parseJSON } from "../google_drive";
+import { ParsedQs } from "qs";
+import { Request, Response } from "express-serve-static-core";
+
+import { encrypt, decrypt } from "../util"
+
+import fetch from "node-fetch";
+import { verifyAdminToken, verifyToken } from "../middlewares/auth.middleware";
+import { json } from "body-parser";
+const router = express.Router()
+
+const required = true;
+
+router.get("/", verifyAdminToken, (req, res) => {
+  res.render("admin", {
+    title: "Task robot admin page",
+    host: req.headers.host,
+  });
+});
+
+const getsurvey = async (query: string | ParsedQs, req: Request<{}>, res: Response<any>) =>  {
+  try {
+    const survey_url = new URL(query["url"]);
+    const worker_id = query["WorkerId"]
```

Comment:
This seems unsafe — if we don't have a `WorkerId` we will run into problems here.

Reply:
Removed query independence.